pinot-core/src/main/java/com/linkedin/pinot/core/operator/CombineGroupByOperator.java

```diff
@@ -156,7 +156,8 @@ public void runJob() {
       boolean opCompleted = operatorLatch.await(_timeOutMs, TimeUnit.MILLISECONDS);
       if (!opCompleted) {
         // If this happens, the broker side should already timed out, just log the error and return
-        String errorMessage = "Timed out while combining group-by results after " + _timeOutMs + "ms";
+        String errorMessage = "Timed out while combining group-by results after " + _timeOutMs + "ms"
+            + " Broker request: " + _brokerRequest;
```

Comment:
Does broker request have a requestId in it? If not, it might be better to stick in there, and just print the requestId.

Reply:
+1 on just adding the request id. Also, can we not find the server that timed out in broker log along with pql and request Id

Reply:
I looked to log just the requestId too, but broker request doesn't have it.

Typically we see a bunch of requests timeout when a bad query hits the server; in a high qps use case, its harder to see exactly what the offending request is. Sometimes, its obvious from the query pattern and sometimes its not so straightforward. This change takes the guess work away.

Reply:
I was proposing to find a way to have requestId available there. Also, I am unsure how this solves the problem when it is not obvious from the query pattern what the offending query is. Moreover, for long queries, it may be hard to match the brokerRequest to its corresponding PQL.

Reply:
It looks like this other change that I made actually gets what we want: https://github.com/linkedin/pinot/pull/3180

That code wasn't deployed to production when the error that I noticed happened.